lib/storage/bucket.js

```diff
@@ -288,8 +306,51 @@ Bucket.prototype.getMetadata = function(callback) {
 };
 
 /**
+ * Restrict access to your bucket's contents to only project owners.
+ *
+ * @param {function=} callback - The callback function.
+ *
+ * @example
+ * bucket.makePrivate(function(err) {});
+ */
+Bucket.prototype.makePrivate = function(callback) {
+  callback = callback || util.noop;
+
+  this.setMetadata({
+    predefinedAcl: 'private'
+  }, function(err) {
+    callback(err || null);
+  });
+};
+
+/**
+ * Make the files in your bucket publicly readable.
+ *
+ * @param {function=} callback - The callback function.
+ *
+ * @example
+ * bucket.makePublic(function(err) {});
+ */
+Bucket.prototype.makePublic = function(callback) {
+  callback = callback || util.noop;
+
+  this.setMetadata({
+    predefinedAcl: 'publicRead'
```

Comment:
+1 to making changes propagate immediately.

Re "list objects" : I don't believe that it means objects are readable... List means GET /bucket will tell you the list of keys, but GET /bucket/objects/object is what we're looking for.

I think `bucket.makePublic` needs to do more than just mark the bucket as public. It should also (depending on arguments):

- set the defaultACL to 'public' (so all future items are made public) (ie, `if (future) { /* set default ACL to public */ }`)
- set all existing items to 'public' (so all existing items are public (ie `if (recursive) { /* set everything to public */ }`)